### PR TITLE
Sort the output of ProjectHealthConsoleReportBuilder

### PR DESCRIPTION
**Pull request by mvegter**

### CODE REVIEW

File: src/main/kotlin/com/autonomousapps/internal/advice/ProjectHealthConsoleReportBuilder.kt [hunk 1]
```diff
@@ -35,9 +36,9 @@ internal class ProjectHealthConsoleReportBuilder(
         shouldPrintNewLine = true
 
         appendReproducibleNewLine("Unused dependencies which should be removed:")
-        val toPrint = removeAdvice.joinToString(separator = "\n") {
-          " ${it.fromConfiguration}(${printableIdentifier(it.coordinates)})"
-        }
+        val toPrint = removeAdvice.mapToOrderedSet {
+          "${it.fromConfiguration}(${printableIdentifier(it.coordinates)})"
+        }.joinToString(prefix = "  ", separator = "\n  ")
```
[the remark this answers is not in the document]
Comment (mvegter) [reply src/main/kotlin/com/autonomousapps/internal/advice/ProjectHealthConsoleReportBuilder.kt:41]: This is actually the same formatting as is used by the v1 AdvicePrinter. On our internal projects (against latest main with v2 model) the output is unsorted, hence the change.

[the remark this answers is not in the document]
Comment (autonomousapps) [reply src/main/kotlin/com/autonomousapps/internal/advice/ProjectHealthConsoleReportBuilder.kt:41]: Is it important for you that there be two spaces prefixing each line? Trying to understand if this is an implicit API.

[the remark this answers is not in the document]
Comment (mvegter) [reply src/main/kotlin/com/autonomousapps/internal/advice/ProjectHealthConsoleReportBuilder.kt:41]: We do not depend on the indent behavior. Personally, I like the two spaces indent as it easier to read and consistent with the v1 version.

**Review by autonomousapps — CHANGES_REQUESTED** [review]
I think I realized why I'm confused. At the top of this init block, there's this:
```kotlin
val dependencyAdvice = projectAdvice.dependencyAdvice
val removeAdvice = mutableSetOf<Advice>()
val addAdvice = mutableSetOf<Advice>()
val changeAdvice = mutableSetOf<Advice>()
val compileOnlyAdvice = mutableSetOf<Advice>()
val processorAdvice = mutableSetOf<Advice>()
```
I'm almost certain that `val dependencyAdvice = projectAdvice.dependencyAdvice` _is_ sorted. However, I break up that large sorted set into a handful of _hash_ sets, which lose the ordering information. If each of those mutable sets were instead `sortedSetOf<Advice>()`, then I think you could drop the `mapToOrderedSet`, which I would prefer. Would you mind testing that?

I don't mind you changing it back to the double-spaced output.

**Review by autonomousapps — APPROVED** [review]
Thanks! LGTM.

**Comment by mvegter** [comment]
@autonomousapps  There is a small regression here, we are sorting on the artifact and not on the suggestion, I guess that is why the initial one was using `mapToOrderSet` instead of the sortedSet

**Comment by autonomousapps** [comment]
Sounds like we need a regression test!

**Comment by autonomousapps** [comment]
I was going to reopen this, but it's a PR and not an issue. Mind filing an issue? I think this situation is amenable to a simpler unit test of this particular class.

**Pull request merged**